app/js/lib/foxbox.js

```diff
@@ -218,78 +246,57 @@ export default class Foxbox extends Service {
       return Promise.resolve();
     }
 
-    return new Promise((resolve, reject) => {
-      let hasChanged = false;
-
-      Promise.all([
-          this.getServices(),
-          fetchJSON(`${this.origin}/services/list`)
-        ])
-        .then(res => {
-          // Detect newly connected/disconnected services.
-          const storedServices = res[0];
-          const fetchedServices = res[1];
-
-          // Any newly connected devices?
-          fetchedServices.some(serviceA => {
-            const service = storedServices.find(serviceB => serviceA.id === serviceB.id);
-            if (!service) {
-              hasChanged = true;
-              return true;
-            }
-          });
-
-          // Any newly disconnected devices?
-          storedServices.some(serviceA => {
-            const service = fetchedServices.find(serviceB => serviceA.id === serviceB.id);
-            if (!service) {
-              hasChanged = true;
-              return true;
-            }
-          });
-
-          return [storedServices, fetchedServices];
-        })
-        .then(res => {
-          // Detect change in service states.
-          const storedServices = res[0];
-          const fetchedServices = res[1];
-
-          Promise.all(fetchedServices.map(service => this.getServiceState(service.id)))
-            .then(states => {
-              fetchedServices.forEach((fetchedService, id) => {
-                // Populate the service objects with states.
-                fetchedService.state = states[id];
-
-                const storedService = storedServices.find(s => s.id === fetchedService.id);
-
-                if (!storedService) {
-                  this._dispatchEvent('service-state-change', fetchedService);
-
-                  // Populate the db with the latest service.
-                  db.setService(fetchedService);
+    let fetchedServicesPromise = fetchJSON(`${this.origin}/services/list`).then(
+      (services) => {
+        // TODO: We should ask for state only for services that actually support
+        // it.
+        return Promise.all(
+          services.map((service) => {
+            // Use empty state if service failed to return actual state.
+            return this.getServiceState(service.id).catch(() => ({})).then(
+              (state) => service.state = state
+            );
+          })
+        ).then(() => services);
+      }
+    );
+
+    return Promise.all([
+      this.getServices(),
+      fetchedServicesPromise
```

Comment:
If I understand correctly, `fetchedServicesPromise` contains value and not a promise. Is that right?
If so, we want to run both promises synchronously in the `Promise.all()`.